internal/querycoordv2/meta/target_manager.go

```diff
@@ -3,184 +3,278 @@ package meta
 import (
 	"sync"
 
+	"github.com/samber/lo"
+	"go.uber.org/zap"
+
 	"github.com/milvus-io/milvus/internal/log"
 	"github.com/milvus-io/milvus/internal/proto/datapb"
-	"github.com/milvus-io/milvus/internal/util/funcutil"
-	"go.uber.org/zap"
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+type TargetScope = int
+
+const (
+	CurrentTarget TargetScope = iota + 1
+	NextTarget
 )
 
 type TargetManager struct {
-	rwmutex sync.RWMutex
+	rwMutex sync.RWMutex
 
-	segments   map[int64]*datapb.SegmentInfo
-	dmChannels map[string]*DmChannel
+	// all read segment/channel operation happens on current -> only current target are visible to outer
+	// all add segment/channel operation happens on next -> changes can only happen on next target
+	// all remove segment/channel operation happens on Both current and next -> delete status should be consistent
+	current *target
+	next    *target
 }
 
 func NewTargetManager() *TargetManager {
 	return &TargetManager{
-		segments:   make(map[int64]*datapb.SegmentInfo),
-		dmChannels: make(map[string]*DmChannel),
+		current: newTarget(),
+		next:    newTarget(),
 	}
 }
 
+func (mgr *TargetManager) UpdateCollectionCurrentTarget(collectionID int64, partitionIDs ...int64) {
+	mgr.rwMutex.Lock()
+	defer mgr.rwMutex.Unlock()
+	log.Info("start to update current target for collection",
+		zap.Int64("collectionID", collectionID),
+		zap.Int64s("PartitionIDs", partitionIDs))
```

Comment:
1. Use log.With() to make sure all logs below would print collectionID and partitionIDs
2. PartitionIDs -> partitionIDs